cdap-api/src/main/java/co/cask/cdap/api/common/DatasetProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.common;
+
+import java.util.Set;
+
+/**
+ * Defines an interface for components which use Datasets.
+ */
+public interface DatasetProvider {
+
+  /**
+   * @return An immutable set of {@link co.cask.cdap.api.dataset.Dataset DataSets} that are used
+   */
+  Set<String> getDatasets();
```

Comment:
The original WorkerSpecification was a Set<String>.  I've also looked at other examples like FlowletDefinition, MapReduceSpecification and their getDatasets all return a Set<String> too.  Do I need to change this for Workers/Custom actions?